src/Kernel32/Kernel32+PROCESS_MEMORY_EXHAUSTION_INFO.cs

```diff
@@ -0,0 +1,86 @@
+{
+    using System;
+
+    /// <content>
+    /// Contains the <see cref="PROCESS_MEMORY_EXHAUSTION_INFO"/> nested type.
+    /// </content>
+    public partial class Kernel32
+    {
+        /// <summary>
+        /// Allows applications to configure a process to terminate if an allocation fails to commit memory.
+        /// This structure is used by the <see cref="PROCESS_INFORMATION_CLASS"/> class.
+        /// </summary>
+        public struct PROCESS_MEMORY_EXHAUSTION_INFO
+        {
+            /// <summary>
+            /// The only valid value for <see cref="PROCESS_MEMORY_EXHAUSTION_INFO.Version"/>
+            /// </summary>
+            public const ushort PME_CURRENT_VERSION = 1;
+
+            /// <summary>
+            /// Value for <see cref="PROCESS_MEMORY_EXHAUSTION_INFO.Value"/> field to disable
+            /// the <see cref="PROCESS_MEMORY_EXHAUSTION_TYPE.PMETypeFailFastOnCommitFailure"/> feature
+            /// </summary>
+            public static UIntPtr PME_FAILFAST_ON_COMMIT_FAIL_DISABLE = UIntPtr.Zero;
+
+            /// <summary>
+            /// Value for <see cref="PROCESS_MEMORY_EXHAUSTION_INFO.Value"/> field to enable
+            /// the <see cref="PROCESS_MEMORY_EXHAUSTION_TYPE.PMETypeFailFastOnCommitFailure"/> feature
+            /// </summary>
+            public static UIntPtr PME_FAILFAST_ON_COMMIT_FAIL_ENABLE = new UIntPtr(0x1);
+
+            /// <summary>
+            /// This should be set to <see cref="PME_CURRENT_VERSION"/>
+            /// </summary>
+#pragma warning disable IDE1006 // Naming Styles
+            public ushort Version;
+#pragma warning restore IDE1006 // Naming Styles
+
+            /// <summary>
+            /// Reserved
+            /// </summary>
+#pragma warning disable IDE1006 // Naming Styles
+            public ushort Reserved;
+#pragma warning restore IDE1006 // Naming Styles
+
+            /// <summary>
+            /// Type should be set to <see cref="PROCESS_MEMORY_EXHAUSTION_TYPE.PMETypeFailFastOnCommitFailure"/> (this is the only
+            /// type available).
+            /// </summary>
+#pragma warning disable IDE1006 // Naming Styles
+            public PROCESS_MEMORY_EXHAUSTION_TYPE Type;
+#pragma warning restore IDE1006 // Naming Styles
+
+            /// <summary>
+            /// Used to turn the feature on or off
+            ///  <list type="table">
+            ///  <listheader><term>Function</term><term>Setting</term></listheader>
+            ///  <item><term>Enable</term><term><see cref="PME_FAILFAST_ON_COMMIT_FAIL_ENABLE"/></term></item>
+            ///  <item><term>Disable</term><term><see cref="PME_FAILFAST_ON_COMMIT_FAIL_ENABLE"/></term></item>
+            ///  </list>
+            /// </summary>
+#pragma warning disable IDE1006 // Naming Styles
+            public UIntPtr Value;
+#pragma warning restore IDE1006 // Naming Styles
+
+            public static PROCESS_MEMORY_EXHAUSTION_INFO Create(PROCESS_MEMORY_EXHAUSTION_TYPE pmeType, bool enable)
```

Comment:
I don't think we have `Create` factories anywhere in the repo that do more than initialize a `cbSize` type field. The spirit of the repo is generally to not add friendlier APIs above the p/invoke ones except where *very* peculiar memory allocation/release requirements are present that would be likely to be done incorrectly by the caller if not done within the library. I feel like this method may not belong.

Of course value is great, but we have to hold a line somewhere or else it'll always be a debate as to whether a particular method or class beyond p/invokes is worth adding, and eventually the repo is full of not only p/invokes but another BCL. Certainly your method here is a far cry from a BCL, but the easiest line to draw and defend is (IMO) on the other side of this method. 😏

Reply:
<< explaining my thought process here - not pushing back >> 

The pattern I inferred from the `cbSize` examples was "If the documentation specifies that the function requires (i.e., *forces*) initialization of one or more structure members to specific values, then provide a *Create* helper to satisfy that contract". 

In this instance, I wanted to provide a `Create` helper that set the `.Version` field to `PME_CURRENT_VERSION`, and `.Type` to `.PMETypeFailFastOnCommitFailure`. Then the `Create` helper grew into a method with parameters to factor in future expansion to `PROCESS_MEMORY_EXHAUSTION_TYPE` enum...  ;-) 

A minimal `Create()` would have simply initialized `.Version` (and maybe `.Reserved`) and nothing else. 

I'll remove.